lib/node/init.ts

```diff
@@ -1,5 +1,33 @@
+// The type of preload script exports.
+interface PreloadExports {
+  onBuiltinModulesPatched?: () => {};
+}
+
+// Get the preload path.
+// Note that the path might be updated later so code should not cache the path.
+const v8Util = process._linkedBinding('electron_common_v8_util');
+const getPreloadPath = () => {
+  return v8Util.getHiddenValue(process, 'preload') as string | undefined;
+};
+
+// Run the preload script.
+const Module = require('module') as NodeJS.ModuleInternal;
+let preload: PreloadExports | undefined;
+{
+  const preloadPath = getPreloadPath();
+  if (preloadPath) {
+    try {
+      // Load the preload script similiar to how Node implements --require flag.
+      const parent = new Module('internal/preload', null);
+      preload = Module._load(preloadPath, parent, false) as PreloadExports | undefined;
+    } catch (error) {
+      console.error(`Error happened when running preload (${preloadPath})`, error);
+    }
+  }
+}
+
 // Initialize ASAR support in fs module.
-import { wrapFsWithAsar } from './asar-fs-wrapper';
+const { wrapFsWithAsar } = require('./asar-fs-wrapper');
```

Comment:
`import { wrapFsWithAsar } = require('./asar-fs-wrapper');`

Reply:
otherwise we would have to do
```ts
import type * as asarFsWrapper from './asar-fs-wrapper';
const { wrapFsWithAsar } = require('./asar-fs-wrapper') as typeof asarFsWrapper;
```

Reply:
The code formatter insists putting the import statements to the first line which kind of makes this PR hard to review, I would like to merge #41148 first and rebase on it to solve this.

Reply:
`// eslint-disable-line import/first`